### DISCUSSION TITLE
Terrain layer exact surface.

### DISCUSSION THREAD

**Discussion by nilscb** [visgl/deck.gl#7266] [Ideas]
It does not seem to be possible to specify terrain layer using all possible triangles.
That is two triangles for every 4 neighboring depth data points. Sometimes it is useful to get an exact surface with more triangles than an approximate surface like for example produced by the delatin algorithm. Does this makes sense and/or are there any plans to implement this?

Thanks
Christian Bonnevie

**Comment by kylebarron** [comment]
Can you set `meshMaxError: 0`?

**Reply by nilscb** [reply]
Good idea but that does not work 100%. If flat areas for instance the algorithm will still produce few triangles.I believe I can find a workaround for it myself. I just wanted to make sure that there where no such plans in your pipeline. Thanks for the input.

**Reply by kylebarron** [reply]
> If flat areas for instance the algorithm will still produce few triangles

But for fully flat areas with no variation in the data... fewer triangles still perfectly represents an exact surface, right?

**Reply by ibgreen** [reply]
> I believe I can find a workaround for it myself

This could be of interest to others... Maybe you would consider contributing your implementation? It should be a very clean change to add a new algorithm to the TerrainLoader in loaders.gl, as it already supports selection of tesselation algorithm based on a string constant option.

**Reply by nilscb** [reply]
I am not a very experienced javascript programmer (have been using c++ mostly) But I will see what I can do :-) In general I guess it is much(!) more straight forward to generate all triangles than what the delatin algorithm does. Also it seems that modern browsers/graphic cards easily handles huge amount of triangles.
